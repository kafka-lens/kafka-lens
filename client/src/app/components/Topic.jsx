import React from 'react';
import PartitionList from './PartitionList';
import '../css/Topic.scss';

const Topic = ({
  id, topicInfo, showPartitions, showMessages,
}) => (
  <div className="topic-header">
    <div
      className="topic-padding"
      id={id}
      topicname={topicInfo.topicName}
      role="button"
      tabIndex="0"
      onKeyPress={showPartitions}
      onClick={showPartitions}
    >
      {topicInfo.topicName}
    </div>
    {topicInfo.showPartitions === true ? (
      <PartitionList showMessages={showMessages} topicInfo={topicInfo} />
    ) : (
      ''
    )}
  </div>
);

export default Topic;
