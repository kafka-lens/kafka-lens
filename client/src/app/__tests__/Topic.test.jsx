import React from 'react';
import { shallow } from 'enzyme';

import Topic from '../components/Topic';

describe('Topic.js unit tests', () => {
  let wrapper;
  const props = {
    id: 1,
    showPartitions: jest.fn(),
    topicInfo: {
      showPartitions: false,
      topicName: 'test1',
      numberOfPartitions: 5,
    },
  };

  const partitionInfoPropWithShowPartitionsTrue = {
    topicInfo: {
      showPartitions: true,
      topicName: 'test1',
      numberOfPartitions: 5,
    },
  };

  beforeAll(() => {
    wrapper = shallow(<Topic {...props} />);
  });

  it('Should render without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call showPartitions on click', () => {
    wrapper
      .find('div')
      .at(0)
      .childAt(0)
      .simulate('click');
    expect(props.showPartitions).toHaveBeenCalled();
  });

  it('Should render a div with inner text "test1" ', () => {
    expect(
      wrapper
        .find('div')
        .at(0)
        .text(),
    ).toEqual('test1');
  });

  it('Should render PartitionList when showMessages = true', () => {
    wrapper.setProps(partitionInfoPropWithShowPartitionsTrue);
    expect(wrapper.exists('PartitionList')).toEqual(true);
  });
});
