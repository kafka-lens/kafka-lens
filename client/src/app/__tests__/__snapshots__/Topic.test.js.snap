// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic.js unit tests Should render without errors 1`] = `
<div
  className="topic-header"
>
  <div
    className="topic-padding"
    id={1}
    onClick={[MockFunction]}
    onKeyPress={[MockFunction]}
    role="button"
    tabIndex="0"
    topicname="test1"
  >
    test1
  </div>
</div>
`;
